docs/features/builtin-buffers.md

```diff
@@ -1,15 +1,42 @@
 # Built-in buffers
+  
+Quix Streams offers high-performance, low-latency buffering capabilities, providing an optimized method for sending and processing time-series data. The buffering mechanism used here helps reduce costs, mainly when dealing with high-frequency data, by improving data compression and reducing serialization and deserialization time.
 
-If you’re sending data at a high frequency, processing time-series data without any buffering involved can be very costly. With smaller messages, data compression cannot be implemented as efficiently. The time spent in serializing and deserializing messages can be improved by sending more values per message.
+Buffers in Quix Streams work at the timestamp level, accumulating timestamps until a specific release condition is met. A packet is then published containing those timestamps and values as a [TimeseriesData](../subscribe.md#timeseriesdata-format) package.
 
-On the other hand, an incorrect buffer strategy can introduce unnecessary latency which may not be acceptable in some use cases.
+ 
+![High level time-series buffering flow](../images/QuixBuffering.png)
 
-Quix Streams provides you with a high performance and low-latency buffer, and a simple way to  configure buffer parameters. This enables you to configure the buffer to balance latency and cost of memeory allocated to buffers.
 
-Buffers in the library work at the timestamp level. A buffer accumulates timestamps until a release condition is met. A packet is then published containing those timestamps and values as a [TimeseriesData](../subscribe.md#timeseriesdata-format) package.
+The buffer can be used to [subscribe](../subscribe.md#using-a-buffer) to and [publish](../publish.md#using-a-buffer) time-series data.
 
-![High level time-series buffering flow](../images/QuixBuffering.png)
+ ## Leading edge buffer
+ In addition to the package size, timespan and timeout buffer configurations, Quix Streams also offers the `LeadingEdgeBuffer` – a buffer that can be especially useful when data received is out-of-order and must be ordered before it is consumed and late arriving data.
```

Comment:
not sure if I understand the english of ... and late arriving data. But maybe I am being thick :D